### PR TITLE
Javascript: Library v1.1

### PR DESCRIPTION
**Pull request by mke2111**
# Library v1.1

**In this project, we implemented a simple library store using vanilla Javascript with the given functionality.**

- Add a new Book.
- Delete an existing book.
- Change the read status of a book.
- Books are stored in the local storage of the browser.

### CODE REVIEW

File: src/JS/app.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+function Book(title, author, pages, edition, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.edition = edition;
+  this.read = !read;
+}
+
+function showForm() {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  form.className = 'show';
+}
+
+function updateLocalStorage() {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  localStorage.setItem('myLibrary', JSON.stringify(myLibrary));
+}
+
+function addNewBooks() {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  if (ftitle.value === '' || fauthor.value === '' || fpages.value === '' || fedition.value === '') {
+    alert('There are data missing, please check it.');
+  } else {
+    const book = new Book(ftitle.value, fauthor.value, fpages.value, fedition.value, fread);
+    myLibrary.push(book);
+    updateLocalStorage();
+  }
+}
+
+function readContent(s) {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  let status;
+  if (myLibrary[s].read === true) {
+    status = 'Read';
+  } else {
+    status = 'Not Read';
+  }
+  return status;
+}
+
+function showBooks() {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    bookDiv.appendChild(cardPage);
+    bookDiv.appendChild(cardEdition);
+    bookDiv.appendChild(cardRead);
+    bookDiv.appendChild(readButton);
+    bookDiv.appendChild(deleteBtn);
+    bookCard.appendChild(bookDiv);
+  }
+}
+
+function getLocalStorage() {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  myLibrary = JSON.parse(localStorage.getItem('myLibrary'));
+
+  if (myLibrary === null) {
+    myLibrary = [];
+  }
+  showBooks();
+}
+
+function deleteBook(book) {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

File: src/JS/app.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  myLibrary = JSON.parse(localStorage.getItem('myLibrary'));
+  myLibrary.splice(book, 1);
+  localStorage.setItem('myLibrary', JSON.stringify(myLibrary));
+  getLocalStorage();
+}
+
+function readStatus(s) {
```
Comment (addod19) [inline src/JS/app.js]: An ES6 arrow function can be implemented here

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @mke2111 and @diazgio, great job on this project :+1: The project works as expected and has all the features of the app :smile: However, there a few comments we need to address before proceeding :slightly_smiling_face: Kindly check the comments below:

- The index file should have 1 element, for example `<div id="root"></div>` where all the other elements are appended to the parent node. It would be great to create a file for the DOM manipulation task
- The screenshot in the README is broken, kindly fix it
- Kindly add instructions about how to run the app locally
- Kindly implement the ES6 arrow function, for example; replace `function name(){}` with `const name = () => {}`
- Kindly implement a modal for our alert messages instead of the `alert()` or `confirm()`


[Optional]
- Kindy fix the input for the number of pages.  The input type for the number of pages should be `type="number"`

Once these are addressed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](linkedin.com/in/daniel-larbi-addo/)

**Comment by diazgio** [comment]
We try to change the UNIQUE issue with the ESlint that we have but crash the entire app, so we decide to keep that line it's the only thing that is missing, we manage to make all the changes that were requested.

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 

Great job guys, thanks for making the required changes.

now you can merge, but please first add this disable line before the import that is causing the linter error: `/* eslint-disable import/extensions */`

## Happy Coding :tophat:

**Pull request merged**